lib/core/src/data/weakset.rs

```diff
@@ -0,0 +1,113 @@
+use std::collections::HashSet;
+use std::rc::{Rc, Weak};
+use std::hash::{Hash, Hasher};
+pub trait Hashable = Hash + PartialEq;
+
+pub struct WeakElem<T : Hashable> {
+    elem : Weak<T>
+}
+
+pub struct WeakRef<T : Hashable> {
+    elem : Rc<T>,
+    owner : Rc<WeakSet<T>>
+}
+
+impl<T : Hashable> Drop for WeakRef<T> {
+    fn drop(&mut self) {
+        unsafe {
+            // no worries because it will happen synchronously
+            Rc::get_mut_unchecked(&mut self.owner).data.remove(&WeakElem { elem : Rc::downgrade(&self.elem) });
```

Comment:
Please fix this and other lines that have >80 chars. Be sure to read our code style guide.

Reply:
weakset.rs was removed as suggested